### PR TITLE
Minor refactor on keychain tests

### PR DESCRIPTION
**Pull request by AsceticMonk**
**Changes**
- Removed some redundant tests
- Added more assertions
- Minor formatting changes

### CODE REVIEW

File: libs/SalesforceSDKCommon/SalesforceSDKCommonTests/KeychainItemManagerTests.swift [hunk 1]
```diff
@@ -26,37 +26,33 @@
...
-        XCTAssertTrue(keychainReadResult.success)
-        XCTAssertNil(keychainReadResult.data)
-        
-        let data = "ATESTSTRING2".data(using: .utf8)  ?? Data()
-        let writeResult = KeychainHelper.write(service: serviceName, data: data, account: accountName)
-        XCTAssertTrue(writeResult.success)
-        XCTAssertNotNil(writeResult.data)
-        
-        let readResult = KeychainHelper.read(service: serviceName, account: accountName)
+        var readResult = KeychainHelper.read(service: serviceName, account: nil)
```
Comment (bbirman) [inline libs/SalesforceSDKCommon/SalesforceSDKCommonTests/KeychainItemManagerTests.swift:36]: why is the account nil?

Comment (AsceticMonk) [reply libs/SalesforceSDKCommon/SalesforceSDKCommonTests/KeychainItemManagerTests.swift]: It's a copy-and-paste error. Fixed, thanks!

File: libs/SalesforceSDKCommon/SalesforceSDKCommonTests/KeychainItemManagerTests.swift [hunk 2]
```diff
@@ -72,138 +68,124 @@ final class KeychainItemManagerTests: XCTestCase {
...
-        XCTAssertTrue(writeResult.success)
-        XCTAssertNotNil(writeResult.data)
-        
-        let readResult = KeychainHelper.read(service: serviceName, account: nil)
+        readResult = KeychainHelper.read(service: serviceName, account: accountName)
         XCTAssertTrue(readResult.success)
-        XCTAssertNotNil(readResult.data)
-        
+        XCTAssertNil(readResult.data)
+        XCTAssertTrue(readResult.status == errSecSuccess)
```
Comment (bbirman) [inline libs/SalesforceSDKCommon/SalesforceSDKCommonTests/KeychainItemManagerTests.swift:82]: possible nit: is there a reason not to use XCTAssertEqual here and other places? I think the failure message would be more descriptive

**Review by bbirman — APPROVED** [review]

File: libs/SalesforceSDKCommon/SalesforceSDKCommonTests/KeychainItemManagerTests.swift (hunk 2 above)
Comment (AsceticMonk) [reply libs/SalesforceSDKCommon/SalesforceSDKCommonTests/KeychainItemManagerTests.swift]: Good call! Done.

**Pull request merged**